independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/Reflections.java

```diff
@@ -148,4 +197,74 @@ public static Object invokeMethod(Class<?> clazz, String name, Class<?>[] paramT
         }
     }
 
+    static final class MethodKey {
+
+        final Class<?> clazz;
+        final String methodName;
+        final Class<?>[] parameterTypes;
```

Comment:
Could you try with:
- removing the finals here and in the other key class
- caching the hashCode in the constructor
- not using `Objects.hash(clazz, methodName)` as it does one more array allocation

That's what I did in HV and it was beneficial. If it doesn't make any significant difference, we can keep it your way but as you have a benchmark and we are optimizing things, it's worth a quick check IMHO

Reply:
I can give it a try!

Reply:
Btw, the remark on `Objects.hash()` is also valid for the Method one.

Reply:
After giving it some thoughts, let's keep the ``final``s and just experiment with the rest.

It's a bit different than in HV as you don't really publish these maps safely before accessing them in a multi-threaded environments.

Reply:
So I've tried to cache the hashCodes and the results are:
```
Benchmark                         Mode  Cnt         Score         Error  Units
ReflectionsBenchmark.findField   thrpt   25  70640655,243 ± 1919015,675  ops/s
ReflectionsBenchmark.findMethod  thrpt   25  57293358,785 ± 1065670,627  ops/s
```
Now I'm going to replace the `Objects.hash()` ...

Reply:
```
Benchmark                         Mode  Cnt          Score         Error  Units
ReflectionsBenchmark.findField   thrpt   25  104686242,346 ± 3802225,528  ops/s
ReflectionsBenchmark.findMethod  thrpt   25   81447915,881 ±  661570,727  ops/s
```
Awesome, thanks for your tips ;-)

Reply:
Yeah, we have a lot of pressure on the memory in HV as the path is really very optimized CPU-wise so we ended up with all sort of tricks to optimize the map lookups :).

Glad it could be of use for another project.